### ISSUE TITLE
Quote this to prevent word splitting

### ISSUE THREAD

**Issue by thisguyshouldworkforus** [koalaman/shellcheck#2662]
#### For bugs
- Rule Id (SC2046): 
- My shellcheck version (shellcheck.exe --version: 0.8.0)
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2046)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

```bash
#!/usr/bin/env bash

read -r -a EXISTING_PATHS <<< $(grep -Eri '^/' /etc/logrotate.conf /etc/logrotate.d/* | sed -r 's/(\s+)?(\{$)?|(\*[\.\_]log\ \{)//g' | sed 's/\/$//g' | awk -F ':' '{print $2}' | grep -Eiv '\.log$' | sort -u)
```

#### Here's what shellcheck currently says:
```bash
#!/usr/bin/env bash

# Error on SC2046
read -r -a EXISTING_PATHS_UNQUOTED <<< $(grep -Eri '^/' /etc/logrotate.conf /etc/logrotate.d/* | sed -r 's/(\s+)?(\{$)?|(\*[\.\_]log\ \{)//g' | sed 's/\/$//g' | awk -F ':' '{print $2}' | grep -Eiv '\.log$' | sort -u)
echo "${#EXISTING_PATHS_UNQUOTED[@]}"
13

# No error on SC2046
read -r -a EXISTING_PATHS_QUOTED <<< "$(grep -Eri '^/' /etc/logrotate.conf /etc/logrotate.d/* | sed -r 's/(\s+)?(\{$)?|(\*[\.\_]log\ \{)//g' | sed 's/\/$//g' | awk -F ':' '{print $2}' | grep -Eiv '\.log$' | sort -u)"
echo "${#EXISTING_PATHS_QUOTED[@]}"
1
```

#### Here's what I wanted or expected to see:
- If I quote this, only 1 item -- the first item -- is populated. I have been "ignoring" this error for years, but I think it should either be addressed in this specific use case, or, if I've been messing this up for years, then my ignorance with BASH should be rightly called out and corrected! Either works for me!  Thanks!

**Comment by brother** [comment]
logrotate states that paths with spaces must be enclosed in quotes, this means that logrotate config files where a path with spaces in it is the first on the line will not be found with your example. If the quoted path is second or later in the line you could happen upon it.

```
brother ~$ head -n 1 /tmp/early /tmp/late 
==> /tmp/early <==
"/tmp/example/file path" /tmp/some/path {

==> /tmp/late <==
/tmp/some/path "/tmp/example/file path" {
brother ~$ grep -Eri '^/' /tmp/early /tmp/late 
/tmp/late:/tmp/some/path "/tmp/example/file path" {
```

Looking at the sed statements I think you are anyhow removing the spaces in the first one there. Given that I think you are safe to just add an ignore statement before the execution of this. 
Example
```
brother ~$ grep -Eri '^/' /tmp/early /tmp/late | sed -r 's/(\s+)?(\{$)?|(\*[\.\_]log\ \{)//g' 
/tmp/late:/tmp/some/path"/tmp/example/filepath"
```

**Comment by TriMoon** [comment]
Why not just pipe the output of that line to the read command, you are already using multiple pipes anyhow :thinking: 
It will make the code more readable for a human also...
_(Plus `shellcheck` would not complain i guess)_
eg:
```bash
grep -Eri '^/' /etc/logrotate.conf /etc/logrotate.d/* \
| sed -r 's/(\s+)?(\{$)?|(\*[\.\_]log\ \{)//g' \
| sed 's/\/$//g' \
| awk -F ':' '{print $2}' \
| grep -Eiv '\.log$' \
| sort -u \
| read -r -a EXISTING_PATHS_UNQUOTED
```

So in your example i must agree with `shellcheck` on this example...
